internal/versions/versions.go

```diff
@@ -92,37 +121,238 @@ func (h *handler) GetMustGatherImages(openshiftVersion, cpuArchitecture, pullSec
 	if err != nil {
 		return nil, err
 	}
-	h.mustGatherVersions[cacheKey]["ocp"] = ocpMustGatherImage
+	h.mustGatherVersions[configKey]["ocp"] = ocpMustGatherImage
 
-	versions := h.mustGatherVersions[cacheKey]
+	versions := h.mustGatherVersions[configKey]
 	return versions, nil
 }
 
 // Returns the default ReleaseImage entity for a specified CPU architecture
-func (h *handler) GetDefaultReleaseImage(cpuArchitecture string) (*models.ReleaseImage, error) {
+func (h *handler) GetDefaultReleaseImageByCPUArchitecture(cpuArchitecture string) (*models.ReleaseImage, error) {
+	if err := h.validateReleaseImageCPUArchitecture(cpuArchitecture); err != nil {
+		return nil, err
+	}
+
+	// First, try from configuration releases
 	defaultReleaseImage := funk.Find(h.releaseImages, func(releaseImage *models.ReleaseImage) bool {
 		return releaseImage.Default && *releaseImage.CPUArchitecture == cpuArchitecture
 	})
 
-	if defaultReleaseImage == nil {
-		return nil, errors.Errorf("Default release image is not available")
+	if defaultReleaseImage != nil {
+		return defaultReleaseImage.(*models.ReleaseImage), nil
 	}
 
-	return defaultReleaseImage.(*models.ReleaseImage), nil
+	// If not found, try from DB releases
+	dbRelease, err := common.GetOpenshiftVersionFromDBWhere(
+		h.db,
+		`"default" = ? AND ? = ANY(cpu_architectures)`,
+		true,
+		cpuArchitecture,
+	)
+	if err != nil {
+		h.log.WithError(err).Debug("error occurred while trying to find the default release in the DB")
+	}
+
+	releaseImage, err := h.createReleaseImageFromDBOpenshiftVersion(dbRelease, cpuArchitecture)
+	if err != nil {
+		h.log.WithError(err).Debugf(
+			"error occurred while trying to create release image from openshift version %s and CPU architecture %s",
+			*dbRelease.DisplayName,
+			cpuArchitecture,
+		)
+		return nil, err
+	}
+
+	if releaseImage != nil {
+		return releaseImage, nil
+	}
+
+	return nil, errors.Errorf("Default release image is not available")
 }
 
-func (h *handler) GetReleaseImage(ctx context.Context, openshiftVersion, cpuArchitecture, pullSecret string) (*models.ReleaseImage, error) {
-	image, err := h.getReleaseImageFromCache(openshiftVersion, cpuArchitecture)
-	if err == nil || h.kubeClient == nil {
-		return image, err
+func (H *handler) getURLForDBRelease(release common.OpenshiftVersion, cpuArchitecture string) *string {
+	mapToURLCpuArchitecture := map[string]string{
+		common.AMD64CPUArchitecture: common.X86CPUArchitecture,
+		common.ARM64CPUArchitecture: common.AARCH64CPUArchitecture,
 	}
 
-	// The image doesn't exist in the cache.
-	// Fetch all the cluster image sets, cache them, then search the cache again
+	arch, ok := mapToURLCpuArchitecture[cpuArchitecture]
+	if !ok {
+		arch = cpuArchitecture
+	}
 
+	return swag.String(fmt.Sprintf(releaseImageURLPrefix+"%s-%s", *release.DisplayName, arch))
+}
+
+func (h *handler) getReleaseSupportLevel(release models.ReleaseImage) (*string, error) {
```

Comment:
I feel like a lot of this belongs with the db-specific code rather than in the versions handler.